torch/distributed/_spmd/distribute.py

```diff
@@ -249,7 +249,7 @@ def _convert_output(
 
         traced_dispatch, result_obj = _build_dummy_add_graph(dt, node_to_obj)
 
-        wait = [n for n in traced_dispatch.graph.nodes if n.name == "wait_comm"]
+        wait = [n for n in traced_dispatch.graph.nodes if n.name == "wait_comm" or n.name == "wait_tensor"]
```

Comment:
if we move to using the new collectives will we need to check for "wait_comm"?

Reply:
Once we move to the new collectives we'd stop using CommTensor in DMesh and we could drop supporting it.

This will require implementing all remaining collectives used by DMesh/DTensor which will take a while.